File: lib/models/bundle.dart
class Bundle {
  final int id;
  final String ispName;
  final String name;
  final String type; // DATA, CALL, SMS
  final String volume;
  final double price;
  final int? validityDays;

  Bundle({
    required this.id,
    required this.ispName,
    required this.name,
    required this.type,
    required this.volume,
    required this.price,
    this.validityDays,
  });

  factory Bundle.fromJson(Map<String, dynamic> json) {
    return Bundle(
      id: json['id'] ?? 0,
      ispName: json['ispName'] ?? '',
      name: json['name'] ?? '',
      type: json['type'] ?? '',
      volume: json['volume'] ?? '',
      price: (json['price'] ?? 0.0).toDouble(),
      validityDays: json['validityDays'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'ispName': ispName,
      'name': name,
      'type': type,
      'volume': volume,
      'price': price,
      'validityDays': validityDays,
    };
  }
}

File: backend/src/main/java/com/switchn/backend/entity/Bundle.java
package com.switchn.backend.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;

@Entity
@Table(name = "bundles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Bundle {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "isp_id", nullable = false)
    private ISP isp;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String type; // DATA, CALL, SMS

    @Column(nullable = false)
    private String volume; // e.g., "1GB", "100 minutes"

    @Column(nullable = false)
    private BigDecimal price;

    @Column(name = "validity_days")
    private Integer validityDays;

    @Column(name = "is_active")
    private Boolean isActive = true;
}
